value/src/main/java/com/google/auto/value/processor/autovalue.vm

```diff
@@ -75,16 +75,14 @@ ${modifiers}class $subclass$formalTypes extends $origClass$actualTypes {
     ## the constructor is called from the extension code.
 
     #if ($identifiers)
-    if ($p == null) {
-      throw new NullPointerException("Null $p.name");
-    }
+        this.$p = `java.util.Objects`.requireNonNull($p, "Null $p");
     #else
-    `java.util.Objects`.requireNonNull($p);
+        this.$p = `java.util.Objects`.requireNonNull($p);
     #end
 
-  #end
-
+  #else
```

Comment:
Given that there is now an `#else`, I might be inclined to invert the sense of the `#if`:

```
#if ($p.kind.primitive || $p.nullable || ($builderTypeName != "" && $isFinal)
## ...comment...
this.$p = $p;
#elseif ($identifiers)
this.$p = `java.util.Objects`.requireNonNull($p, "Null $p");
#else
this.$p = `java.util.Objects`.requireNonNull($p);
#end
```

Reply:
Yes, it looks better that way.